src/hardware/gus.cpp

```diff
@@ -997,11 +1078,9 @@ bool Gus::PerformDmaTransfer()
 	}
 
 	if (dma_channel->has_reached_terminal_count) {
-		// We've hit the terminal count, so enable that bit
-		dma_ctrl |= DMA_TC_STATUS_BITMASK;
+		dma_control_register.SetPendingTerminalCountIrq();
 
-		// Raise the TC irq if needed
-		if ((dma_ctrl & 0x20) != 0) {
+		if (dma_control_register.WantsIrqOnTerminalCount()) {
```

Comment:
This is great with the descriptive method names—top notch stuff 👍🏻

Reply:
Thanks! It's wordy :laughing: but self documenting (and matches the docs description), so :man_shrugging: :smile: